src/Mod/Part/App/PropertyTopoShapeList.cpp

```diff
@@ -0,0 +1,193 @@
+ *                                                                         *
+ ***************************************************************************/
+
+#include "PreCompiled.h"
+
+#include <Base/Console.h>
+#include <Base/Reader.h>
+#include <Base/Writer.h>
+
+#include "TopoShapePy.h"
+#include "Part2DObject.h"
+
+#include "PropertyTopoShapeList.h"
+
+using namespace App;
+using namespace Base;
+using namespace std;
+using namespace Part;
+
+
+//**************************************************************************
+// PropertyTopoShapeList
+//++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+TYPESYSTEM_SOURCE(Part::PropertyTopoShapeList, App::PropertyLists)
+
+//**************************************************************************
+// Construction/Destruction
+
+PropertyTopoShapeList::PropertyTopoShapeList()
+{
+
+}
+
+PropertyTopoShapeList::~PropertyTopoShapeList()
+{
+    for (std::vector<TopoShape*>::iterator it = _lValueList.begin(); it != _lValueList.end(); ++it)
+        if (*it) delete *it;
+}
+
+void PropertyTopoShapeList::setSize(int newSize)
+{
+    for (unsigned int i = newSize; i < _lValueList.size(); i++)
+        delete _lValueList[i];
+    _lValueList.resize(newSize);
+}
+
+int PropertyTopoShapeList::getSize() const
+{
+    return static_cast<int>(_lValueList.size());
+}
+
+void PropertyTopoShapeList::setValue(TopoShape* lValue)
+{
+    if (lValue) {
+        aboutToSetValue();
+        _lValueList.resize(1);
+        _lValueList[0] = lValue;
+        hasSetValue();
+    }
+}
+
+void PropertyTopoShapeList::setValues(const std::vector<TopoShape*>& lValue)
+{
+    aboutToSetValue();
+    _lValueList.resize(lValue.size());
+    for (unsigned int i = 0; i < lValue.size(); i++)
+        _lValueList[i] = lValue[i];
+    hasSetValue();
+}
+
+PyObject *PropertyTopoShapeList::getPyObject()
+{
+    PyObject* list = PyList_New(getSize());
+    for (int i = 0; i < getSize(); i++)
+        PyList_SetItem( list, i, _lValueList[i]->getPyObject());
+    return list;
+}
+
+
+void PropertyTopoShapeList::setPyObject(PyObject *value)
+{
+    // check container of this property to notify about changes
+    Part2DObject* part2d = dynamic_cast<Part2DObject*>(this->getContainer());
+
+    if (PySequence_Check(value)) {
+        Py_ssize_t nSize = PySequence_Size(value);
+        std::vector<TopoShape*> values;
+        values.resize(nSize);
+
+        for (Py_ssize_t i=0; i < nSize; ++i) {
+            PyObject* item = PySequence_GetItem(value, i);
+            if (!PyObject_TypeCheck(item, &(TopoShapePy::Type))) {
+                std::string error = std::string("types in list must be 'Shape', not ");
+                error += item->ob_type->tp_name;
+                throw Base::TypeError(error);
+            }
+
+            values[i] = static_cast<TopoShapePy*>(item)->getTopoShapePtr();
+        }
+
+        setValues(values);
+        if (part2d)
```

Comment:
I don't think that this acceptGeometry() stuff is needed here. It looks like a hack copied over from PropertyGeometryList.

Reply:
Removed.